src/app/components/week-selector/week-selector.component.spec.ts

```diff
@@ -0,0 +1,123 @@
+    spyOn(component.passDate, 'emit');
+ 
+    fixture.detectChanges();
+    let currentWeekMinusOne = getWeek(subWeeks(component.date, 1))
+
+
+    component.ngOnInit()
+    component.onPreviousWeek()
+
+    expect(component.weekNumber).toBe(currentWeekMinusOne);
+    expect(component.passDate.emit).toHaveBeenCalledWith(currentWeekMinusOne)
+    expect(component.disabledLeft).toBe(true)
+    expect(component.disabledRight).toBe(false)
+  });
+
+  it('should emit weeknumber plus one when clicked', () => {
+    spyOn(component.passDate, 'emit');
+    fixture.detectChanges();
+    let currentWeekPlussOne = getWeek(addWeeks(component.date, 1))
+    component.isCurrentWeek = true;
+    component.date = component.currentDate
+
+    component.ngOnInit()
+    component.onNextWeek()
+
+    expect(component.weekNumber).toBe(currentWeekPlussOne);
+    expect(component.passDate.emit).toHaveBeenCalledWith(currentWeekPlussOne)
+    expect(component.disabledLeft).toBe(false)
+    expect(component.disabledRight).toBe(false)
+  });
+});
+
+describe('WeekSelectorComponent current date before start date', () => {
+  let component: WeekSelectorComponent;
+  let fixture: ComponentFixture<WeekSelectorComponent>;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      declarations: [ WeekSelectorComponent ]
+    })
+    .compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(WeekSelectorComponent);
+    component = fixture.componentInstance;
+    component.currentDate = new Date(getWeek(subWeeks(Date.now(), 1)));
+    component.startDate = new Date(Date.now())
+
+    fixture.detectChanges();
+  });
+
+  it('should emit weeknumber minus one when clicked', () => {
+    spyOn(component.passDate, 'emit');
+ 
+    fixture.detectChanges();
+    let currentWeekMinusOne = getWeek(subWeeks(component.date, 1))
+
+    component.ngOnInit()
+    component.onPreviousWeek()
+
+    expect(component.weekNumber).toBe(currentWeekMinusOne);
+    expect(component.passDate.emit).toHaveBeenCalledWith(currentWeekMinusOne)
+    expect(component.disabledLeft).toBe(true)
+    expect(component.disabledRight).toBe(false)
+  });
+});
+
+describe('WeekSelectorComponent current date equal to current week', () => {
+  let component: WeekSelectorComponent;
+  let fixture: ComponentFixture<WeekSelectorComponent>;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      declarations: [ WeekSelectorComponent ]
+    })
+    .compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(WeekSelectorComponent);
+    component = fixture.componentInstance;
+    component.startDate = new Date(Date.now());
+    component.currentDate = new Date(Date.now());
+
+    fixture.detectChanges();
+  });
+
+  it('should emit weeknumber minus one when clicked', () => {
+    component.ngOnInit()
+    
+    expect(component.disabledLeft).toBe(false)
+    expect(component.disabledRight).toBe(true)
+  });
+});
```

Comment:
Ik zie hier heel veel herhaalde code (DRY). Misschien goed om te weten dat je describe-blocks kunt nesten. Je kunt dan in een nested beforeEach bijvoorbeeld `component.startDate` en `component.currentDate` voor deze laatste test instellen op dezelfde datum, maar je gebruikt het component dat op je line 18 aanmaakt

Reply:
> Ik heb een paar dingetjes aangestipt, ik ben benieuwd of je er iets mee kunt!

Zeker! Leerzame en waardevolle feedback. Super relaxed dat je altijd te tijd neemt om dingen goed te reviewen. Goede eigenschap!

Reply:
Je hebt helemaal gelijk. Ik zie dat ik veel te veel code gecopy paste heb in deze test

Reply:
Ach, het is al positief dat je uberhaupt tests schrijft. Testen vind ik een van de moeilijkste dingen in Angular; het wordt heel snel heel ingewikkeld.